### ISSUE TITLE
Cannot read property "headers" of undefined

### ISSUE THREAD

**Issue by justinmchase** [tmpvar/jsdom#1182]
I just pulled down your latest update and I'm still getting the issue with the xhr and cookies :(

I completely agree that this is a bug in node-xhr but they appear to be very unresponsive with PR's. I'm basically using cucumber.js, jsdom, angular together. Here is the error:

```
TypeError: Cannot read property 'headers' of undefined
  at [object Object].getAllResponseHeaders (e:\code\evolve\evolve-client-core\node_modules\jsdom\node_modules\xmlhttprequest\lib\XMLHttpRequest.js:232:27)
  at [object Object].requestLoaded [as onload] (e:\code\evolve\evolve-client-core\node_modules\angular\angular.js:10480:17)
  at [object Object].dispatchEvent (e:\code\evolve\evolve-client-core\node_modules\jsdom\node_modules\xmlhttprequest\lib\XMLHttpRequest.js:572:25)
  at setState (e:\code\evolve\evolve-client-core\node_modules\jsdom\node_modules\xmlhttprequest\lib\XMLHttpRequest.js:595:14)
  at e:\code\evolve\evolve-client-core\node_modules\jsdom\node_modules\xmlhttprequest\lib\XMLHttpRequest.js:308:13
  at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)
```

And here is the specific code in angular.js that is triggering the error:
```
xhr.onload = function requestLoaded() {
  var statusText = xhr.statusText || '';

  // responseText is the old-school way of retrieving response (supported by IE8 & 9)
  // response/responseType properties were introduced in XHR Level2 spec (supported by IE10)
  var response = ('response' in xhr) ? xhr.response : xhr.responseText;

  // normalize IE9 bug (http://bugs.jquery.com/ticket/1450)
  var status = xhr.status === 1223 ? 204 : xhr.status;

  // fix status code when it is 0 (0 status is undocumented).
  // Occurs when accessing file resources or on Android 4.1 stock browser
  // while retrieving files from application cache.
  if (status === 0) {
    status = response ? 200 : urlResolve(url).protocol == 'file' ? 404 : 0;
  }

  completeRequest(callback,
      status,
      response,
      xhr.getAllResponseHeaders(), // error here
      statusText);
};
```

Now I'm pretty surprised by this because I tried out your PR and ran the unit tests and confirmed that it seemed to not trigger this error but it seems like angular is doing something specifically a little differently that's unexpected.

I will debug it more a little later and see if I can't find a more elegant PR. Otherwise... and I'm reticent to do this but may have to fork that node-xhr project so we can fix some bugs in there because this is painful.

Sorry and thank you :(

**Comment by domenic** [comment]
Well we didn't test `getAllResponseHeaders()`, just `getResponseHeader`. That might be the specific problem. Oops.

**Comment by domenic** [comment]
Anyway I think the best PR (besides forking node-xhr, which if you do, thank you!!) would be to monkey-patch xhr.getAllResponseHeaders like we already do xhr.open and xhr.send.

**Comment by justinmchase** [comment]
re: `getAllResponseHeaders` vs `getResponseHeader` shoot, I didn't catch that. My bad.

**Comment by domenic** [comment]
Should be fixed in 6.0.1; happy to reopen if it isn't!

**Comment by justinmchase** [comment]
Yup thanks!
